File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Event.java
package com.faendir.lightning_launcher.multitool.proxy;

/**
 * @author lukas
 * @since 04.07.18
 */
public interface Event extends Proxy {
    Item getItem();

    String getSource();

    long getDate();

    Container getContainer();

    Screen getScreen();

    String getData();

    float getTouchX();

    float getTouchY();

    float getTouchScreenX();

    float getTouchScreenY();
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Event.kt
package com.faendir.lightning_launcher.multitool.proxy

/**
 * @author lukas
 * @since 04.07.18
 */
interface Event : Proxy {
    val item: Item?

    val source: String

    val date: Long

    val container: Container

    val screen: Screen

    val data: String

    val touchX: Float

    val touchY: Float

    val touchScreenX: Float

    val touchScreenY: Float
}
